app/models/visit.rb

```diff
@@ -40,4 +47,41 @@ def adult?(visitor)
   def same_visit?(other)
     other.visit_id == self.visit_id
   end
+
+  def log_any_missing_information
+    prefix = "Visit #{visit_id} is missing a"
```

Comment:
We can tidy the logging implementation up a bit here. We are consistently asking `prisoner` if its `blank?` - we should try and consolidate those calls down to one.

Also think we should extract all of this logging stuff out into an `Visit::IntegrityChecker`(or similar) object that can be used in the `ApplicationController`.